### PR TITLE
Terms without ids

### PR DESCRIPTION
**Pull request by glassjoseph**
Closes issue #55.

### CODE REVIEW

File: src/snac/server/database/DBUtil.php [hunk 1]
```diff
@@ -4239,9 +4288,9 @@ private function savePlace($vhInfo, $cObj, $relatedTable, $fkID)
                                                    $gObj->getID(),
                                                    $this->db->boolToPg($gObj->getConfirmed()),
                                                    $gObj->getOriginal(),
-                                                   $this->thingID($gObj->getGeoTerm()),
-                                                   $this->thingID($gObj->getType()),
-                                                   $this->thingID($gObj->getRole()),
+                                                   $this->termID($gObj->getGeoTerm()),
```
Comment (deternitydx) [inline src/snac/server/database/DBUtil.php]: Does GeoTerm work the same way as Term?  They're slightly different structures, so we might want to keep thingID here?

File: src/snac/server/database/SQL.php [hunk 2]
```diff
@@ -4034,6 +4034,51 @@ public function selectTerm($termID)
...
+     */
+    public function selectTermByValueAndType($value, $type)
+    {
+        $qq = 's_term_by_value';
+        $this->sdb->prepare($qq,
+                            'select
+                            id, type, value, uri, description
+                            from vocabulary where value=$1 and type=$2');
+        $result = $this->sdb->execute($qq, array($value, $type));
+        $row = $this->sdb->fetchrow($result);
```
Comment (deternitydx) [inline src/snac/server/database/SQL.php]: What does this do if a matching term isn't found?  Is row just null?  I'm curious if we think there could be two terms that match, but I'm doubtful that would be the case.

File: src/snac/server/database/SQL.php [hunk 3]
```diff
@@ -4034,6 +4034,51 @@ public function selectTerm($termID)
...
+     */
+    public function selectTermByUri($uri)
+    {
+        $qq = 's_term_by_uri';
+        $this->sdb->prepare($qq,
+                            'select
+                            id, type, value, uri, description
+                            from vocabulary where uri=$1');
+        $result = $this->sdb->execute($qq, array($uri));
+        $row = $this->sdb->fetchrow($result);
```
Comment (deternitydx) [inline src/snac/server/database/SQL.php]: Same as above.  I'm wondering now if selectTerm() had a provision for what happens if the ID isn't found.

File: src/snac/server/database/DBUtil.php (hunk 1 above)
Comment (glassjoseph) [reply src/snac/server/database/DBUtil.php]: Geoterm has some differences. I wrote an isEmpty() function for it to mirror Term, but I agree it's safer and less coupled just to use thingID().

File: src/snac/server/database/SQL.php (hunk 3 above)
Comment (glassjoseph) [reply src/snac/server/database/SQL.php]: All three selectTerm..() functions return an empty row if the match isn't found. Right now, that's caught in DBUtil populateTerm() line 1718. What would be the best way to catch that in the SQL?

Comment (deternitydx) [reply src/snac/server/database/SQL.php]: I think I overthought.  fetchRow in DatabaseConnector just uses pg_fetch_all in PHP, which returns false if there is no row.  So, my concern of what happens when we try to get a row that doesn't exist isn't well founded.  Leaving it as is works well, sorry for the confusion.

**Pull request merged**